lib/fluent/plugin/in_tail.rb

```diff
@@ -669,6 +710,51 @@ def on_timer
       end
     end
 
+    class TailThread
+      class Pool
+        def initialize(max_size, &session)
+          @max_size = max_size
+          @queue = Queue.new
+          @threads = []
+          @running = false
+          session.call(self)
+          @running = true
+        ensure
+          stop
```

Comment:
It will not clean up because `@running` is `false` while running `session.call(self)`, and `stop` doesn't nothing when `@running` is `false`.

Reply:
I tried to handle more properly to terminate `@queue`.
But I not sure this patch https://github.com/fluent/fluentd/pull/3185/commits/e20d0f30e6844a10f05363bb81f5f10ba6d7c0f6 is appropriate for our use case.

Reply:
I decided to use `@queue` instance status to check whether thread pool is running or not:
https://github.com/fluent/fluentd/pull/3185/commits/41aa0cb0b193f9da0d477d2d57b36bd22b39ae5f